### ISSUE TITLE
Multiple changes to IS can cause later change ignored

### ISSUE THREAD

**Issue by Jackie-Jiang** [apache/helix#2309]
### Describe the bug
When multiple changes are made to IS in a short period of time (within the same processing cycle), some changes might be mistakenly ignored. I can see IS different from EV, but no state transition message being sent. This is the same as #331, which seems still not completely fixed.

### To Reproduce
Post 2 IS changes in a short period of time (e.g. add a partition, then remove it). EV might end up in the state of the first IS change, and stop proceeding to the final IS state.

### Expected behavior
EV should match the final state of the IS.

### Additional context
We are running CUSTOM mode, Helix-1.0.4

**Comment by Jackie-Jiang** [comment]
@junkaixue @jiajunwang Could you help take a look if the issue I described can still happen after the fix #363?

**Comment by junkaixue** [comment]
Hmm... Any logs for that? It was hard to reproduce.

**Comment by junkaixue** [comment]
Yeah. It is possible to miss some notification. Long term to resolve it, Helix should enable persistent watcher for all the watchers. It is in our road map.
